server/routes/prisons/exclusionDates/exclusionDatesController.test.ts

```diff
@@ -0,0 +1,126 @@
+import type { Express } from 'express'
+import request from 'supertest'
+import * as cheerio from 'cheerio'
+import { FieldValidationError } from 'express-validator'
+
+import { appWithAllRoutes, flashProvider } from '../../testutils/appSetup'
+import { createMockPrisonService, createMockSessionTemplateService } from '../../../services/testutils/mocks'
+import TestData from '../../testutils/testData'
+import { FlashErrorMessage } from '../../../@types/visits-admin'
+
+let app: Express
+let flashData: Record<string, string | FlashErrorMessage>
+
+const prisonService = createMockPrisonService()
+const sessionTemplateService = createMockSessionTemplateService()
+
+const prisonNames = TestData.prisonNames()
+const excludeDates = ['2023-12-25', '2024-12-25']
+const prison = TestData.prison()
+const prisonWithExcludeDates = TestData.prison({ excludeDates })
+const prisonName = prisonNames[prison.code]
+
+beforeEach(() => {
+  flashData = {}
+  flashProvider.mockImplementation(key => flashData[key])
+
+  prisonService.getPrison.mockResolvedValue({ prison, prisonName })
+
+  app = appWithAllRoutes({ services: { prisonService, sessionTemplateService } })
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('Show excluded dates', () => {
+  it('GET /prisons/{:prisonId}/exclusion-dates', () => {
```

Comment:
Doesn't look like any of the tests for the exclusion dates listing pages (`GET /prisons/{prisonCode}/exclusion-dates`) actually check that the exclusion dates get rendered as a table listing?

Reply:
It's checking if the page is rendered, I added more test, to check excluded Dates table and message if excludedDates array is empty.